core/src/main/scala/cats/Eval.scala

```diff
@@ -310,6 +310,30 @@ private[cats] trait EvalInstances extends EvalInstances0 {
       }
     }
 
+  implicit val catsReducibleForEval: Reducible[Eval] =
```

Comment:
It also occurs to me that a big difference between this `Traverse` instance and the instances that you've added is that yours shouldn't ever lead to stack overflows while this one is probably pretty likely to bite people with them.